cs/cli/vw_prediction.cpp

```diff
@@ -93,21 +93,38 @@ namespace VW
         }
         CATCHRETHROW
 
-            if (length > Int32::MaxValue)
-            {
-                throw gcnew ArgumentOutOfRangeException("Multi-label predictions too large");
-            }
+        if (length > Int32::MaxValue)
+            throw gcnew ArgumentOutOfRangeException("Multi-label predictions too large");
 
         auto values = gcnew cli::array<int>((int)length);
 
         if (length > 0)
-        {
             Marshal::Copy(IntPtr(labels), values, 0, (int)length);
-        }
 
         return values;
     }
 
+    cli::array<ActionScore>^ VowpalWabbitActionScorePredictionFactory::Create(vw* vw, example* ex)
+    {
+#if _DEBUG
+      if (ex == nullptr)
+        throw gcnew ArgumentNullException("ex");
+#endif
+
+      auto length = ex->pred.a_s.size();
+      auto values = gcnew cli::array<ActionScore>(length);
+
+      auto index = 0;
+      for (auto& as : ex->pred.a_s)
+      {
+          values[index].Action = as.action;
+          values[index].Score = as.score;
+          index++;
+      }
```

Comment:
Actually not. So the weird thing is that ‘for each’ modifies the ex->pred.a_s._begin. Not sure why. I think one is only supposed to use it for IEnumerable. That’s the reason I referred to it as magic operator ☺

From: Luong Hoang [mailto:notifications@github.com]
Sent: Friday, June 3, 2016 9:04 AM
To: JohnLangford/vowpal_wabbit <vowpal_wabbit@noreply.github.com>
Cc: Markus Cozowicz <marcozo@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [JohnLangford/vowpal_wabbit] Action/score fix (#1004)


In cs/cli/vw_prediction.cpp<https://github.com/JohnLangford/vowpal_wabbit/pull/1004#discussion_r65703127>:

> +    {

> +#if _DEBUG

> +      if (ex == nullptr)

> +        throw gcnew ArgumentNullException("ex");

> +#endif

> +

> +      auto length = ex->pred.a_s.size();

> +      auto values = gcnew cli::array<ActionScore>(length);

> +

> +      auto index = 0;

> +      for (auto& as : ex->pred.a_s)

> +      {

> +          values[index].Action = as.action;

> +          values[index].Score = as.score;

> +          index++;

> +      }

if you used 'for each' here it should be ok since you're not modifying 'as', right?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/JohnLangford/vowpal_wabbit/pull/1004/files/1914b96baf3f5f88d9746a43e4973d95eaf59e3c#r65703127>, or mute the thread<https://github.com/notifications/unsubscribe/AJrCTtL842-A7h92tXxcm7HZ2KzWj6eeks5qICYqgaJpZM4ItLS6>.